conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala

```diff
@@ -50,16 +52,34 @@ object DynamicInterpreter {
                 experimentalMappingRule: Boolean,
                 enableControlFramework: Boolean,
                 jobShortName: String = "Conformance"
-               ) (implicit spark: SparkSession, dao: EnceladusDAO, progArgs: CmdConfig): DataFrame = {
+               )(implicit spark: SparkSession, dao: EnceladusDAO, progArgs: CmdConfig): DataFrame = {
+
+    implicit val interpreterContext: InterpreterContext = InterpreterContext(inputDf.schema, conformance,
+      experimentalMappingRule, enableControlFramework, jobShortName, spark, dao, progArgs)
+
+    applyCheckpoint(inputDf, "Start")
+
+    val conformedDf = applyConformanceRules(ensureErrorColumnExists(inputDf))
+
+    applyCheckpoint(inputDf, "End")
+    logExecutionPlan(conformedDf)
+
+    conformedDf
+  }
+
+  private def applyConformanceRules(inputDf: DataFrame)
+                                   (implicit ictx: InterpreterContext): DataFrame = {
+    implicit val spark: SparkSession = ictx.spark
+    implicit val dao: EnceladusDAO = ictx.dao
+    implicit val progArgs: CmdConfig = ictx.progArgs
     implicit val udfLib: UDFLibrary = new UDFLibrary
 
-    if (enableControlFramework) {
-      inputDf.setCheckpoint(s"$jobShortName - Start")
-    }
-    val steps = conformance.conformance.sortBy(_.order)
-    val (explodedDf, explodeContext) = prepareDataFrame(inputDf, steps,experimentalMappingRule)
+    var explodeContext = ExplosionContext()
+
+    val steps = getConformanceSteps
```

Comment:
This is no longer called on `InterpreterContext` but here. So you have duplicity in methods.

Reply:
Not sure I follow. Please elaborate, I'm happy to fix.

Reply:
My whole project didn't get updated so I saw the method `getConformanceSteps` as duplicity. My fault. Disregards the comment.
Now I properly see what you did and I guess you moved the `getConformanceSteps` to `DynamicInterpreter` because it has dependencies on methods in the `DynamicInterpreter` and having it in the `InterpreterContext` would introduce unwanted and hard to handle dependencies. Is that correct?

Reply:
Yes, exactly